src/Microsoft.AspNet.Server.Kestrel/KestrelServerInformation.cs

```diff
@@ -39,5 +39,34 @@ private static ICollection<string> GetAddresses(IConfiguration configuration)
 
             return addresses;
         }
+
+        private static int GetThreadCount(IConfiguration configuration)
+        {
+            var threadCountString = configuration["server.threadCount"];
+
+            int threadCount;
+            if (string.IsNullOrEmpty(threadCountString) || !int.TryParse(threadCountString, out threadCount))
```

Comment:
I guess it makes sense to range check things as early as possible, but it's already checked before it's used (since it can be overriden in `Configure`. What do you think? Fail fast?